File: backend/app/src/main/java/com/onair/hearit/app/advertisement/dto/AdvertisementResponse.java
package com.onair.hearit.app.advertisement.dto;

import com.onair.hearit.core.domain.Advertisement;

public record AdvertisementResponse(
        Long id,
        String imageUrl,
        String linkUrl,
        String title
) {
    public static AdvertisementResponse from(Advertisement advertisement) {
        return new AdvertisementResponse(
                advertisement.getId(),
                advertisement.getImageUrl(),
                advertisement.getLinkUrl(),
                advertisement.getTitle()
        );
    }
}

File: android/app/src/main/java/com/onair/hearit/data/dto/AdvertisementResponse.kt
package com.onair.hearit.data.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class AdvertisementResponse(
    @SerialName("id")
    val id: Long,
    @SerialName("imageUrl")
    val imageUrl: String,
    @SerialName("linkUrl")
    val linkUrl: String,
    @SerialName("title")
    val title: String,
)
